hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationBackupHandler.java

```diff
@@ -188,14 +218,21 @@ private Data getBackupOperationData(BackupAwareOperation backupAwareOp) {
         // if getServiceName() method is overridden to return the same name
         // then this will have no effect.
         backupOp.setServiceName(op.getServiceName());
-        return nodeEngine.getSerializationService().toData(backupOp);
+        return backupOp;
     }
 
-    private Backup newBackup(BackupAwareOperation backupAwareOp, Data backupOpData, long[] replicaVersions,
+    private Backup newBackup(BackupAwareOperation backupAwareOp, Object backupOp, long[] replicaVersions,
             int replicaIndex, boolean respondBack) {
-
         Operation op = (Operation) backupAwareOp;
-        Backup backup = new Backup(backupOpData, op.getCallerAddress(), replicaVersions, respondBack);
+        Backup backup;
+        if (backupOp instanceof Operation) {
+            backup = new Backup((Operation) backupOp, op.getCallerAddress(), replicaVersions, respondBack);
```

Comment:
You mean, constructor of `Backup` takes `Object` type in a single constructor instead of two different constructor? Then it decides which one will be used, `Data` or `Operation`